### ISSUE TITLE
[CI] `bk://:book: LinkCheck` is failing/flaky on master.

### ISSUE THREAD

**Issue by rickyyx** [ray-project/ray#34928]
- 7071a4f3c380dcec144a900fd8349a8b181d5a01 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3699#0187d95f-20db-4b0d-91f9-3d75fb3f29c3)
- 59822432c701eac4ef73f61f8978510c2f77345d FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3698#0187d93c-3c56-476e-b16a-7aee714a3265)
- 2e0f1575d584412ee131ca89359c9840d67ba30a FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3683#0187ce93-62ee-49bb-b0df-128028caa5f6)
- 991b5e85eac70f7df91b8aeee2c585592171a6f4 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3682#0187ca99-a0f3-4ffb-95d7-bfecda3b1075)
- daa6df020abc992b27f070ba9298584a578d62e5 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3681#0187ca75-cdcf-4693-b508-d8f517eb9f82)
- 2d1db6ddbd348902a5960b4e65de3048a22297af FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3680#0187ca64-6816-4afd-8a96-886790a0deec)
- d22602a5915cb5b7641071f862c5cb4e62019ee0 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3679#0187ca46-15b6-479a-b6cf-fa50586ad97c)
- ff36b8e7e7f71397fc5352acd827f7f6f9aa4cf4 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3678#0187ca00-5eb8-44b7-8848-359ab3847fcb)
- 23d7e9fcfacf8d5cd238526b2a9bbced9fb84f73 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3677#0187c9d6-7e37-4321-aaf6-ca412cd95b24)
- 407bf98b008fda96b4fff17556943affe2069c29 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3676#0187c9d5-8e79-4a78-9a75-42447d1d84f2)
- c4b67f4932c25a42739c89f5b0e19b282cdd71df FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3675#0187c9d0-b1a5-4119-8759-41544501c27f)
- 1274f865982b6c70220397fe4dca09631e378432 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3674#0187c9d0-722b-49d5-9a30-49688b4a121e)
- dd0dcae865ee775afe609b2b221c2e99e2f8369f FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3672#0187c9a7-e068-47e9-9ee6-7edccfd7b5c3)
- 000eda1dda83c90a161464bf363d30734d2cf9c6 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3671#0187c988-c4a3-445f-9b8f-5e287ca9be9b)
- 0d59be7f27454200c9a2c153ad14b0ad00d541e5 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3670#0187c8ff-3493-4b21-9d66-6dfdf2e59b54)
- 7401b39720b095b302b63cb5bc22af71d803aa79 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3669#0187c863-b51b-455c-9ede-3cbdf6e03162)
- 92f4fe725442a74705fe0c39d4cdf3f2d23d77c9 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3668#0187c80a-8021-4078-a453-209258363fcc)
- 922e2c8c95182b64740a5e16b51fe498c47832f9 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3667#0187c80a-7f38-47f0-b5fa-5641dde8862d)
- 86fe469b4ea776d7712c433227afc5885a0a3f24 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3666#0187c712-4d28-4111-8214-867a726098e5)
- 3139d4fcf1a98a421da109cf44770280e7d9e90e FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3665#0187c6be-3fc4-45ee-a373-c8d9d354b43e)
- 239fe5b09a8245f04323e1aec1e4757129d07e90 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3664#0187c61c-079f-4fa4-95d3-ac0a63493364)
- 8211f86cd3472f7842078f9fc62465602ab49a4c FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3663#0187c5a3-8263-4d97-87f5-5de9a30affdf)
- 0954f816444eb7b1b758cd441d48dc50da1eb87f FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3662#0187c54d-95a4-47cd-acc5-571e64496dbc)
- b8386b3341e443c0d7302f89be081cf14f0f158d FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3661#0187c54b-53ae-4a22-a426-6792ddf2e9ab)
- 3ac7ba186db0baf27b4f030971ac835fc0b3e12f FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3660#0187c54b-5080-406e-b3d9-5815ca552899)
- b9db3f458f6415737607a2e3f7692ccc7c9a0d18 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3659#0187c511-9b46-4df9-9987-75b48f55b75e)
- f682777c8a657e36e13be63aca59a993e536c3fb FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3658#0187c4c5-18de-49da-ae01-5957ca9cdcb6)
- c30eb864f5fcf0988ec9848449293afe80a75051 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3657#0187c492-0ce8-4059-b6b7-3619f602f024)
- f419788daf1be7b5608689f23756ddb6187c340f FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3656#0187c47e-da83-4e79-ab4c-97d847bb7ff4)
- 4b0445744503a9edb2e34e358526e58ea2c7b53c FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3655#0187c463-1df5-48fd-b8ec-de4ef721d77f)
- 136c74068b7c7bf50c4ed043dde21ffe0ce8ecc2 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3593#0187b54e-5787-4d4f-904d-a4ed2465aa9b)
- a150463b15fc22b21787891e894ede6a59ebb554 FAILED [:book: LinkCheck](https://buildkite.com/ray-project/oss-ci-build-branch/builds/3592#0187b51f-5541-4b40-94ee-efb00666b8a9)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-bk://:book: LinkCheck-END
....

**Comment by rickyyx** [comment]
Looks like it started with https://github.com/ray-project/ray/commit/c4a29b2bdfca165149228caff22679dc5e8ce939 ?

**Comment by raulchen** [comment]
I think @maxpumperla is fixing the issue. re-assigning to him.

**Comment by anyscalesam** [comment]
this is still all red for the last couple of hundred of runs and has been open for more than half a year now. i propose jailing this and marking this ticket as a p1. we have bigger fish to fry.

@maxpumperla @rickyyx agree?

**Comment by rickyyx** [comment]
Is this still the right lint job to look at tho? @can-anyscale

**Comment by rickyyx** [comment]
> @rickyyx yes, this is a doc-specific lint check; still the right job to look at and has been failing for half a year. devprod is moving docinfra into its scope so we suppose to look at this. But agree on the P1.

Sounds good - i am assigning this back to you then? LMK if you need me to own this.

**Comment by can-anyscale** [comment]
Sound good!

**Comment by architkulkarni** [comment]
Linkcheck is failing on the release branch. E.g. https://buildkite.com/ray-project/oss-ci-build-branch/builds/7224#018c5fda-9f97-404d-944f-5ea40ecd290b/500-922

```

ERROR: Cannot install Pygments==2.13.0 and Pygments==2.16.1 because these package versions have conflicting dependencies.
--
  |  
  | The conflict is caused by:
  | The user requested Pygments==2.13.0
  | The user requested Pygments==2.16.1
  | The user requested (constraint) pygments==2.13.0
  |  
  | To fix this you could try to:
  | 1. loosen the range of package versions you've specified
  | 2. remove package versions to allow pip attempt to solve the dependency conflict
  |  
  | ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

Based on the above, it looks like the setup fails so the job doesn't run.  

@can-anyscale is this a release blocker, or do we just ignore the linkcheck job for the 2.9.0 release?

**Comment by can-anyscale** [comment]
CC: @angelinalg @peytondmurray do you want to fix linkcheck for 2.9.0? the cut is this Friday
